OPHD/States/CrimeRateUpdate.cpp

```diff
@@ -3,13 +3,24 @@
 #include "../Things/Structures/Structure.h"
 #include "../StructureManager.h"
 #include <NAS2D/Utility.h>
+#include <random>
+#include <functional>
+
+
+namespace {
+	std::random_device randomDevice;
+	std::mt19937 generator(randomDevice());
+	std::uniform_int_distribution<int> distribution(0, 1000);
+	auto random = std::bind(distribution, std::ref(generator));
+}
```

Comment:
I believe I once expressed some concern in the past about putting random number code in individual modules. I don't remember the exact reason. I believe it was something to do with reproducibility, or how this might impact saved games, or game recordings, or multiplayer if that is ever implemented. If random numbers are generated and used in the same order on all clients, then there is no need to synchronize multiplayer clients for each value. Instead, a random number state can be shared once at the beginning, and then all clients will generate the same values from there. Similar for game recordings that make use of random numbers. You can store the initial state once, and then generate the same sequence during playback, without having to record the outcome of every random number generated.

It may be a bit early to worry about that design aspect though. In particular, I don't have any suggestion for how to improve this. I guess at some point we might need to develop some high level policy on how random numbers should be generated and used by the game engine.

Reply:
Agreed that this is a bit of a tough one mostly from the aspect of multiplayer. I've considered a multiplayer aspect but when I really think about how it would be done in this game, I'm not sure it would actually be 'fun' so I haven't done much with it.

That stated, this is fine for me. There's a bit of magic numbering going on here, something I've tried to mitigate or at least note when they show up. Besides that this is fine.

Reply:
I would like to merge without resolving this issue. I did open issue #974 to track. I may be able to take a stab at resolving this Random Number Generator issue after this PR closes.

I would approach it as unifying the location of the random number generator in the code base more so than prepping for multiplayer.